index.js

```diff
@@ -24,106 +25,68 @@ app.get('/info', (req, res) => {
 
 // route for displaying all products
 app.get('/products/all', (req, res) => {
-
-  const jsonOutputObject = {
-    products: []
-  };
-
-  for (let product of productData.products) {
-
-    // Deep copying to prevent changes in original JSON object in next step
-    const productToBeInserted = JSON.parse(JSON.stringify(product));
-
-    // Adding category name to product
-    for (let category of categoryData.categories) {
-      if (product.categoryId === category.id) {
-        productToBeInserted.categoryName = category.categoryName;
-        break;
-      }
-    }
-
-    jsonOutputObject.products.push(productToBeInserted);
-  }
-
-  res.send(jsonOutputObject);
+  res.send({ products: dataService.getCombinedProductMap });
 });
 
 // Getting product info from its ID
 app.get('/product/:id', (req, res) => {
 
   const userInput = req.params.id;
   let productId;
-  let jsonOutputObject = null;
+  let productToBeReturned;
 
   // Generating valid ID and returning error message when ID is invalid
   if (userInput < 1) {
     res.send('Invalid product ID');
     return;
   } else {
-    productId = `P${userInput.padStart(3, '0')}`
+    productId = `P${userInput.padStart(3, '0')}`;
   }
 
-  for (let product of productData.products) {
+  for (product of dataService.getCombinedProductMap) {
     if (product.id === productId) {
-      jsonOutputObject = product;
+      productToBeReturned = product;
       break;
     }
   }
 
-  // Checking for null at the end would cause NullPointerException when trying to get the categoryName
-  if (jsonOutputObject == null) {
+  if (productToBeReturned == null) {
```

Comment:
Why not 
```suggestion
  if (productToBeReturned === null) {
```
or 
```suggestion
  if (!productToBeReturned) {
```
?

Reply:
The second suggestation seems better. Done!